signprot/views.py

```diff
@@ -423,6 +424,165 @@ def GProtein(request, dataset="GuideToPharma", render_part="both"):
+        context['tree_class_b1_options']['anchor'] = 'tree_class_b1'
+        context['tree_class_b1_options']['branch_trunc'] = 60
+        context['tree_class_b1_options']['label_free'] = [1,]
+        context['tree_class_b1'] = json.dumps(class_b1_data.get_nodes_dict(None))
+        class_b2_data = tree.get_tree_data(ProteinFamily.objects.get(name__startswith='Class B2 (Adhesion)'))
+        context['tree_class_b2_options'] = deepcopy(tree.d3_options)
+        context['tree_class_b2_options']['anchor'] = 'tree_class_b2'
+        context['tree_class_b2_options']['label_free'] = [1,]
+        context['tree_class_b2'] = json.dumps(class_b2_data.get_nodes_dict(None))
+        class_c_data = tree.get_tree_data(ProteinFamily.objects.get(name__startswith='Class C (Glutamate)'))
+        context['tree_class_c_options'] = deepcopy(tree.d3_options)
+        context['tree_class_c_options']['anchor'] = 'tree_class_c'
+        context['tree_class_c_options']['branch_trunc'] = 50
+        context['tree_class_c_options']['label_free'] = [1,]
+        context['tree_class_c'] = json.dumps(class_c_data.get_nodes_dict(None))
+        class_f_data = tree.get_tree_data(ProteinFamily.objects.get(name__startswith='Class F (Frizzled)'))
+        context['tree_class_f_options'] = deepcopy(tree.d3_options)
+        context['tree_class_f_options']['anchor'] = 'tree_class_f'
+        context['tree_class_f_options']['label_free'] = [1,]
+        context['tree_class_f'] = json.dumps(class_f_data.get_nodes_dict(None))
+        class_t2_data = tree.get_tree_data(ProteinFamily.objects.get(name='Class T (Taste 2)'))
+        context['tree_class_t2_options'] = deepcopy(tree.d3_options)
+        context['tree_class_t2_options']['anchor'] = 'tree_class_t2'
+        context['tree_class_t2_options']['label_free'] = [1,]
+        context['tree_class_t2'] = json.dumps(class_t2_data.get_nodes_dict(None))
+        # end copied section from StructureStatistics View
+        slug_translate = {'001': "ClassA", '002': "ClassB1", '004': "ClassC", '006': "ClassF"}
+        selectivitydata = {}
+        selectivitydata_gtp_plus = {}
+        receptor_dictionary = []
+        for slug in slug_translate.keys():
+            jsondata = {}
+            jsondata_gtp_plus = {}
+            for gp in gproteins:
+                # Collect GTP
+                gtp_couplings = list(ProteinGProteinPair.objects.filter(protein__family__slug__startswith=slug, source="GuideToPharma", g_protein=gp)\
+                                .order_by("protein__entry_name")\
+                                .values_list("protein__entry_name", flat=True)\
+                                .distinct())
+
+                # Other coupling data with logmaxec50 greater than 0
+                # TODO - coupling source should not be hardcoded
+                other_couplings = list(ProteinGProteinPair.objects.filter(protein__family__slug__startswith=slug, source__in=["Aska2", "Bouvier2"])\
```

Comment:
Just a heads-up.
Aska2 and Bouvier2 don't exist any longer in the latest version.
They are now just Inoue and Bouvier.

Reply:
Thanks for the heads-up just fixed it with an exclude call instead of the other way round.